lib/pbench/server/auth/auth.py

```diff
@@ -54,7 +54,11 @@ def encode_auth_token(self, time_delta: datetime.timedelta, user_id: int) -> str
 
     def get_secret_key(self):
         try:
-            return os.getenv("SECRET_KEY", "my_precious")
+            return (
+                OpenIDClient.PUBLIC_KEY
+                if OpenIDClient.PUBLIC_KEY
+                else os.getenv("SECRET_KEY", "my_precious")
+            )
         except Exception as e:
             Auth.logger.exception("Error {} getting JWT secret", e)
```

Comment:
And, while we're on the topic, what does this `try` block accomplish?  Is there anything in there which could raise an exception?...and, if there is, will the caller accept a value of `None` being returned??

Reply:
You are right, there is nothing here that could raise an Exception and it will always return some value. If nothing is set it will return `my_precious` as a key.